### PR TITLE
Add issue template

### PR DESCRIPTION
**Pull request by jeddy3**
Ref: https://github.com/stylelint/stylelint/issues/762

I’ve add the issue template (in a friendly question and answer format).

I don’t think there’s any need for a pull request template at the moment and I think it's best if we avoid putting up barriers to *code* contributions.

I also haven’t moved `CONTRIBUTING.md` out of the root as, I think, it’s a bit different to the templates—in that we currently use it to suggest ways in which a reader can help out. As such, I’d rather keep it easily findable in the root. Doing so will also keep things simple when generating the website.

### CODE REVIEW

**Comment by ntwb** [comment]
Adding the `.github` folder to `.eslintignore` should fix the 4 ESLint errors report by CI tests:

```sh
> eslint . --ignore-path .gitignore && remark . --quiet --frail
.github/ISSUE_TEMPLATE.md
        1:1     warning  Do not use `_` in a file name                  no-file-name-irregular-characters
       20:1     warning  Remove 1 line before node                      no-consecutive-blank-lines
       39:1     warning  Remove 1 line before node                      no-consecutive-blank-lines
  50:49-50:100  warning  Don’t use literal URLs without angle brackets  no-literal-urls
⚠ 4 warnings
```

**Comment by ntwb** [comment]
Here's an example of a GitHUb issue template implemented today:
• https://github.com/TCPR/Thermos/issues/new

My concerns here are:
• Will this be seen as just a *wall of text* that will be confusing to reporters/contributors?
• Is it easy for issue reporters/contributors to clearly see which blocks of text to replace with their own?

_*I think what GitHub has done is a good start in addressing this issue, though I'd rather see these customizations shown as a type of "markdown form" where the current unchanged "CONTRIBUTING.md" content is displayed rather than inline within the primary form textbox.*_

**Comment by jeddy3** [comment]
> Will this be seen as just a wall of text that will be confusing to reporters/contributors?

Yep, I had the same concerns. Hence trying a simple question and answer template and using the minimum amount of markdown for it.

File: .github/ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+> Describe the issue. Is it a bug, a rule enhancement, or a new rule?
```
Comment (davidtheclark) [inline .github/ISSUE_TEMPLATE.md]: How about: "Is it a bug or a feature request (new rule, new option, etc.)?"

File: .github/ISSUE_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+e.g.
+
+```css
+.foo {
+  color: pink;
+}
+```
+
+> What stylelint configuration is need to reproduce this issue?
```
Comment (davidtheclark) [inline .github/ISSUE_TEMPLATE.md]: is needed

**Comment by davidtheclark** [comment]
Cool. I'm curious to see how this works.

Here are some other questions that could be useful. They could be added like the others (with an example), or maybe added to a list "Additional helpful info" or something at the bottom, if we don't want too many questions (or we could leave them out).

- Which version of stylelint are you using?
- How are you running stylelint: CLI, PostCSS plugin, Node API?
- Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

**Comment by davidtheclark** [comment]
Also, @jeddy3 : Did something go wrong with the latest release? I don't see any notes in the release list -- seems weird.

File: .github/ISSUE_TEMPLATE.md (hunk 1 above)
Comment (jeddy3) [inline .github/ISSUE_TEMPLATE.md]: Better. Will change.

File: .github/ISSUE_TEMPLATE.md (hunk 2 above)
Comment (jeddy3) [inline .github/ISSUE_TEMPLATE.md]: Will change.

**Comment by jeddy3** [comment]
> Here are some other questions that could be useful

Those are all good questions. I'll add them in.

> Also, @jeddy3 : Did something go wrong with the latest release?

Oops, my bad. I messed up the version number in the changelog. I've just fixed the release notes.

**Comment by ntwb** [comment]
Looking good :+1:

**Comment by davidtheclark** [comment]
I'm good with this if everyone else is. @jeddy3 feel free to merge when you feel finished.

**Pull request merged**